pdcupdater/handlers/depchain/rpms.py

```diff
@@ -0,0 +1,267 @@
+""" Handlers for an RPM dependency chain model.
+
+The idea here is that these are two of what will eventually be many handlers
+that build and maintain a dependency chain in PDC.
+
+There are two handlers here.  They both respond to koji 'tag' events. One of
+them figures out what the build required at *build* time, and updates PDC with
+that. The other figures out what the build requires at *run* time, and updates
+PDC with that.
+
+"""
+
+import logging
+import time
+
+import pdcupdater.handlers
+import pdcupdater.services
+from pdcupdater.utils import (
+    tag2release,
+    interesting_tags,
+)
+
+
+log = logging.getLogger(__name__)
+
+
+class BaseRPMDepChainHandler(pdcupdater.handlers.BaseHandler):
+    """ Abstract base class. """
+
+    # A list of the types of relationships this thing manages.
+    # This needs to be overridden by subclasses of this base class.
+    managed_types = None
+
+    def __init__(self, *args, **kwargs):
+        super(BaseRPMDepChainHandler, self).__init__(*args, **kwargs)
+        self.koji_url = self.config['pdcupdater.koji_url']
+
+    @property
+    def topic_suffixes(self):
+        return ['buildsys.tag']
+
+    def can_handle(self, msg):
+        if not any([msg['topic'].endswith(suffix)
+                    for suffix in self.topic_suffixes]):
+            return False
+
+        # Ignore secondary arches for now
+        if msg['msg']['instance'] != 'primary':
+            log.debug("From %r.  Skipping." % (msg['msg']['instance']))
+            return False
+
+        interesting = interesting_tags()
+        tag = msg['msg']['tag']
+
+        if tag not in interesting:
+            log.debug("%r not in %r.  Skipping."  % (tag, interesting))
+            return False
+
+        return True
+
+    def _yield_pdc_relationships(self, pdc, release_id):
+        for relationship_type in self.managed_types:
+            entries = pdc.get_paged(
+                pdc['release-component-relationships']._,
```

Comment:
You can do this:
pdc['release-component-relationships/']

Instead of:
pdc['release-component-relationships']._

Not a big deal but it's a little clearer.